src/input.cpp

```diff
@@ -654,11 +654,19 @@ void input_manager::init_keycode_mapping()
     add_gamepad_keycode_pair( JOY_29,        translate_marker_context( "key name", "JOY_29" ) );
     add_gamepad_keycode_pair( JOY_30,        translate_marker_context( "key name", "JOY_30" ) );
 
-    add_mouse_keycode_pair( MOUSE_BUTTON_LEFT,  translate_marker_context( "key name", "MOUSE_LEFT" ) );
-    add_mouse_keycode_pair( MOUSE_BUTTON_RIGHT, translate_marker_context( "key name", "MOUSE_RIGHT" ) );
-    add_mouse_keycode_pair( SCROLLWHEEL_UP,     translate_marker_context( "key name", "SCROLL_UP" ) );
-    add_mouse_keycode_pair( SCROLLWHEEL_DOWN,   translate_marker_context( "key name", "SCROLL_DOWN" ) );
-    add_mouse_keycode_pair( MOUSE_MOVE,         translate_marker_context( "key name", "MOUSE_MOVE" ) );
+    static const std::vector<std::pair<MouseInput, std::string>> mouse_keycode_pairs = {
```

Comment:
Why create a vector containing constants and iterate over it instead of calling the function `add_mouse_keycode_pair` directly (like it was before)?

Worst case: the memory of the vector is now allocated and considered to be in used for the while program runtime.

I see this pattern is used above, so it's probably for a reason, only which reason?

Reply:
That's a fair point, thanks!
The only reason I did that is to both please astyle and to keep the code clean with oneliners.

The approach below does not pass astyle "max characters per line" check. 
```
add_mouse_keycode_pair( MouseInput::LeftButtonPressed, translate_marker_context( "key name", "MOUSE_LEFT_PRESSED" ) );
add_mouse_keycode_pair( MouseInput::LeftButtonReleased, translate_marker_context( "key name", "MOUSE_LEFT" ) );
add_mouse_keycode_pair( MouseInput::RightButtonPressed, translate_marker_context( "key name", "MOUSE_RIGHT_PRESSED" ) );
```

astyle suggests to split the line into two

```
add_mouse_keycode_pair( 
   MouseInput::LeftButtonPressed, translate_marker_context( "key name", "MOUSE_LEFT_PRESSED" ) );
add_mouse_keycode_pair( 
   MouseInput::LeftButtonReleased, translate_marker_context( "key name", "MOUSE_LEFT" ) );
add_mouse_keycode_pair( 
   MouseInput::RightButtonPressed, translate_marker_context( "key name", "MOUSE_RIGHT_PRESSED" ) );

// or

add_mouse_keycode_pair( MouseInput::LeftButtonPressed, 
   translate_marker_context( "key name", "MOUSE_LEFT_PRESSED" ) );
add_mouse_keycode_pair( MouseInput::LeftButtonReleased, 
   translate_marker_context( "key name", "MOUSE_LEFT" ) );
add_mouse_keycode_pair( MouseInput::RightButtonPressed, 
   translate_marker_context( "key name", "MOUSE_RIGHT_PRESSED" ) );

// or even

add_mouse_keycode_pair( 
   MouseInput::LeftButtonPressed, 
   translate_marker_context( "key name", "MOUSE_LEFT_PRESSED" )
 );
add_mouse_keycode_pair( 
   MouseInput::LeftButtonReleased, 
   translate_marker_context( "key name", "MOUSE_LEFT" ) 
);
add_mouse_keycode_pair( 
   MouseInput::RightButtonPressed, 
   translate_marker_context( "key name", "MOUSE_RIGHT_PRESSED" ) 
);
```

Those approaches look ugly and make code harder to read. I was looking for a way to keep "one line" instructions for add_mouse_keycode_pair(). I saw the pattern with vector above and reused it to improve code readability.

But you are right, in terms of performance this a waste of memory

Reply:
@Qrox could you please explain what's the purpose of static vector keyboard_code_keycode_pair? Blame shows you did that change

Reply:
@BevapDin I've refactored it to call ```add_mouse_keycode_pair()``` directly without a static vector. Thanks for the review!

Reply:
> what's the purpose of static vector keyboard_code_keycode_pair

Probably the same reason as yours, and also to save me from typing the function name over and over again. It could definitely be improved by using a `std::array` or something to reduce the memory footprint, but it seemed to not matter that much because the function is only called once and I was too lazy to look up how to do it.